pkg/querier/block.go

```diff
@@ -108,9 +112,9 @@ func (b *blocksQuerier) Select(_ bool, sp *storage.SelectHints, matchers ...*lab
 		return nil, nil, promql.ErrStorage{Err: err}
 	}
 
-	return &blockQuerierSeriesSet{
-		series: series,
-	}, warnings, nil
+	level.Debug(log).Log("series", len(series), "warnings", len(warnings))
+
+	return newTracingSet(b.ctx, "blocksQuerier.seriesSet", &blockQuerierSeriesSet{series: series}), warnings, nil
```

Comment:
We should pass `ctx` instead of `b.ctx` here.

_Unrelated to this PR: renaming `ctx` to `spanCtx` could help making it more clear._

Reply:
> We should pass ctx instead of b.ctx here.

Ignore this comment too, for the same reason.